malloc/malloc.c

```diff
@@ -12,7 +12,13 @@
 */
 char *string_dup(char *src)
 {
-
+    int len = string_length(src);
+    char *dup = malloc(len+1);
+    for (int i = 0; i < len; i++) {
+        *(dup+i) = *(src+i);
```

Comment:
just pointing out here that `*(dup+i)` and `dup[i]`. but of course, use whichever you prefer